hw-3-solution

JlusS opened the pull request:
added new classes Excavator, Bulldozer, Truck with super class Machine.
Implemented doWork() and stopWork() methods in each class

JlusS committed:
added new classes Excavator, Bulldozer, Truck with super class Machine.
Implemented doWork() and stopWork() methods in each class

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Bulldozer.java:
```diff
@@ -0,0 +1,16 @@
+    @Override
+    public boolean doWork() {
+        System.out.println("Bulldozer started his work");
+        return false;
```
The method `doWork()` returns `false`. If the checklist or base class expects a different return value (such as `true` for successful completion), this may be an issue. Please verify the requirements.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Bulldozer.java:
```diff
@@ -0,0 +1,16 @@
+
+    @Override
+    public boolean stopWork() {
+        System.out.println("Bulldozer stoped his work");
```
There is a typo in the output string: "stoped" should be "stopped". This is a minor issue, but correcting it will improve clarity.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Bulldozer.java:
```diff
@@ -0,0 +1,16 @@
+    @Override
+    public boolean stopWork() {
+        System.out.println("Bulldozer stoped his work");
+        return false;
```
The method `stopWork()` returns `false`. If the checklist or base class expects a different return value (such as `true` for successful completion), this may be an issue. Please verify the requirements.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Excavator.java:
```diff
@@ -0,0 +1,16 @@
+    @Override
+    public boolean doWork() {
+        System.out.println("Excavator started his work");
+        return false;
```
The method `doWork()` returns `false`. If the checklist or base class expects a different return value (such as `true` for successful completion), this may be an issue. Please verify the requirements.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Excavator.java:
```diff
@@ -0,0 +1,16 @@
+
+    @Override
+    public boolean stopWork() {
+        System.out.println("Excavator stoped his work");
```
There is a typo in the output string: "stoped" should be "stopped". Please correct this minor issue.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Excavator.java:
```diff
@@ -0,0 +1,16 @@
+    @Override
+    public boolean stopWork() {
+        System.out.println("Excavator stoped his work");
+        return false;
```
The method `stopWork()` returns `false`. If the checklist or base class expects a different return value (such as `true` for successful completion), this may be an issue. Please verify the requirements.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/MainApp.java:
```diff
@@ -1,5 +1,17 @@
+        Machine[] machines = new Machine[]{excavator, bulldozer, truck};
+
+        for (Machine machine : machines) {
+            System.out.println(excavator.doWork());
```
Logic error: The loop should call `machine.doWork()` instead of `excavator.doWork()`. As written, it always calls the `doWork()` method of the `Excavator` instance, not the current machine in the loop. Please update this to ensure each machine's `doWork()` method is called.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Truck.java:
```diff
@@ -0,0 +1,16 @@
+    @Override
+    public boolean doWork() {
+        System.out.println("Truck started his work");
+        return false;
```
The method `doWork()` returns `false`. If the checklist or base class expects a different return value (such as `true` for successful completion), this may be an issue. Please verify the requirements.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Truck.java:
```diff
@@ -0,0 +1,16 @@
+
+    @Override
+    public boolean stopWork() {
+        System.out.println("Truck stoped his work");
```
There is a typo in the output string: "stoped" should be "stopped". Please correct this minor issue.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Truck.java:
```diff
@@ -0,0 +1,16 @@
+    @Override
+    public boolean stopWork() {
+        System.out.println("Truck stoped his work");
+        return false;
```
The method `stopWork()` returns `false`. If the checklist or base class expects a different return value (such as `true` for successful completion), this may be an issue. Please verify the requirements.

mateacademy-ai-mentor reviewed (changes requested):
### Review Summary

Thank you for your submission! There are a few critical issues that need to be addressed before your code can be approved:

1. **Incorrect Return Values:** In all subclasses (`Bulldozer`, `Excavator`, and `Truck`), both `doWork()` and `stopWork()` methods currently return `false`. If the checklist or base class expects these methods to return `true` upon successful completion (which is standard for such methods), this does not meet the core requirements. Please update these methods to return the correct value as per the requirements.

2. **Typo in Output Strings:** There is a typo in the output string of the `stopWork()` method in all subclasses: `"stoped"` should be `"stopped"`. While this is a minor issue, it is specifically mentioned in the checklist and should be corrected to avoid confusion and to meet the output expectations.

3. **Logic Error in MainApp:** In your main application logic, the loop that is supposed to demonstrate polymorphism is incorrectly implemented. Instead of calling `doWork()` and `stopWork()` on each `machine` in the array, it always calls these methods on the `Excavator` instance. This breaks the demonstration of polymorphism and does not fulfill the task requirements. Please update the loop to call the methods on the current `machine` in the array.

Once you fix these issues, your code will meet all the core requirements and will be ready for approval! Keep up the good work, and don't hesitate to reach out if you have any questions. 👍

***

✨ Please rate this review in the Platform's notification center[^1]. I will be grateful for your feedback! ✨

[^1]: [Rate AI review example](https://snipboard.io/gD6XKF.jpg)

> [!IMPORTANT]
> Fix the issues, commit the changes, and then re-request my review using the **`Re-request review` button**[^2] in the Reviewers tab.

[^2]: [Re-request AI review example](https://snipboard.io/nOXxHN.jpg)

> [!TIP]
> If you have any comments about this review, please leave a message in the [Add a comment](#new_comment_field) section below and **mention me with the `@mateacademy-ai-mentor` tag**[^3].
> I will read your message during my next review of your pull request. Please do not reply directly to my comments, as that feature is not supported yet.

[^3]: [Add AI review comments example](https://snipboard.io/hVlOey.jpg)

> [!CAUTION]
> Do not close the pull request until the review is complete. Closing it early can lead to unexpected behavior and disrupt the review process.

JlusS committed:
resolved all problems:
1. stoped -> stopped word in method stopWork();
2. MainApp line 12 correct method run
3. return values in machines classes will return true.

mateacademy-ai-mentor reviewed (approved):
### Review Summary

Great job! 🎉 Your code fully meets all the core requirements and constraints of the task. All classes and methods are correctly implemented, the output is accurate, and polymorphism is demonstrated as expected. There are no critical issues or missing functionalities. Keep up the excellent work!

***

✨ Please rate this review in the Platform's notification center[^1]. I will be grateful for your feedback! ✨

[^1]: [Rate AI review example](https://snipboard.io/gD6XKF.jpg)
